backend/internal/server/server.go

```diff
@@ -48,10 +48,20 @@ func Start(cfg config.Config) (stop func(), err error) {
 	// Mount sync API under /api/sync/v1
 	mux.Mount("/api/sync/v1", syncapi.Handler(syncapi.NewSyncAPIServer(syncService)))
 
-	// Serve static files from embedded frontend directory at root route
-	// This must come after all API routes to ensure proper precedence
-	slog.Info("Serving static files from embedded frontend")
-	mux.Handle("/*", frontend.Handler())
+    // Serve static files and client routes from embedded frontend
+    // This must come after all API routes to ensure proper precedence
+    slog.Info("Serving static files from embedded frontend")
+    mux.Handle("/*", frontend.NewStaticHandler(frontend.RouteConfig{
+        StaticPrefixes: []string{"/dist/"},
+        APIPrefixes:    []string{"/api/"},
+        ClientRoutes: []string{
+            "/",
+            "/components",
+            "/components/*",
+            "/settings",
+            "/sync",
```

Comment:
I don't want to update that everytime. Let's use some routing priorities to handle this. `api` and `dist` first and `/*` is static.